Improve Dataset Support

Myoldmopar opened the pull request:
- [x] Add CI input processing to help validate the inputs of dataset files
- [x] Update the version number comments of the dataset files to V X.Y+ so that we don't have to force change them unless the inputs change
- [x] Try out typedef'd test fixture for datasets to help organize unit test results
- [x] Address dataset unit test failure: Chillers  -- Ran the Chillers file through transition, passes now
- [x] Address dataset unit test failure: PrecipitationSchedulesUSA_fullset_Release  -- Not an actual dataset file, just plain text.  Removed it from the unit test list
- [x] Address dataset unit test failure: RefrigerationCases  -- Not packaged with installer, appears out of date, removed
- [x] Address dataset unit test failure: RefrigerationCasesSetNumberOne  -- Not packaged with installer, appears out of date, removed
- [x] Address dataset unit test failure: ResidentialACsAndHPsPerfCurves  -- The objects in this dataset are full-on stubs missing many required fields.  ~~I'm removing the unit test from this until we decide to either flush out the objects, or comment the stubbed objects out.~~ I updated the dataset file to comment out, but leave in, the generic stubbed out coil objects. The curves are still being processed by the input processor.

Myoldmopar committed:
Add initial dataset input processing support on CI

Myoldmopar committed:
Flush out dataset tests to all dataset idf files; code work is complete

Myoldmopar commented on tst/EnergyPlus/unit/Datasets.unit.cc:
```diff
@@ -0,0 +1,159 @@
+
+	typedef EnergyPlusFixture DataSetFixture;
+
+	std::vector<std::string> getAllLinesInFile( std::string filePath ) {
```
@mbadams5 Did you already implement a way to pass a file path to the process_idf function?  I thought you had, but I couldn't find the declaration.  If not, is there a better way to get the lines of the file to pass into process_idf?

mbadams5 replied:
@Myoldmopar No, but it would be trivially easy to implement. And it would be better than having another function read the IDF (which process_idf already does). I didn't implement this to begin with because I didn't want it to get abused. We can talk about it this week.

Myoldmopar committed:
Test invalid data set inputs

Should be 1 failure that I will revert.

Myoldmopar committed:
Actually add the air cooled chiller test that I expected to fail...

Myoldmopar committed:
Fix unit tests to work properly; revert aircooled chiller to passing

Myoldmopar committed:
Step back from having to configure the dataset source;
Thanks to @macumber for helping guide me to the single configured file of functions

Myoldmopar commented:
:smile: Look at those out-of-date dataset files causing unit test failures @mjwitte ... this will be extremely helpful.  I'm transitioning up those dataset files now and will push them up to make it all green again.

Myoldmopar commented:
Ha! PrecipitationSchedulesUSA_fullset_Release failing...

mjwitte commented:
So, why this route rather than just always run these through transition prior to each release?

Myoldmopar commented:
Well, for one thing, I'm trying to get far away from running bulk transitions of all our files because of the manual changes that get made within a release cycle.  If I then re-run transition on those files then those manual changes get wiped out.  

Secondly, this will trigger a failure **immediately** if the idd changes on one of these objects included in the dataset files, so the dataset files will be updated before merge.  No having to go back and remember anything.  I'm pumped about this.

Myoldmopar committed:
Make some updates to clean up all but one unit test failure

Myoldmopar committed:
Address final unit test failure; this is ready for merge

Myoldmopar committed:
Update version number comments

Myoldmopar committed:
Re-enable the res AC and HP unit test after cleaning up dataset file

Myoldmopar commented:
OK, this is ready for merge.  Now who....

mjwitte commented:
I will review.

mjwitte commented on datasets/ResidentialACsAndHPsPerfCurves.idf:
```diff
@@ -36,21 +36,23 @@
-   ACCoolingEIRFTemp,       !- Energy Input Ratio Function of Temperature Curve Name
-   ACCoolingEIRFFF,         !- Energy Input Ratio Function of Flow Fraction Curve Name
-   ACCoolingPLFFPLR;        !- Part Load Fraction Correlation Curve Name
+! The coil objects in this idf are entered generically with many required fields purposely removed.  These objects are left in the file as comments to show how to use the curves defined in this file, which remain uncommented to allow including the curves in a particular idf, by macro, for example.
```
@Myoldmopar Looks like this one needs this comment?
! The objects and syntax in this dataset comply with EnergyPlus, Version 8.4.0 and above, unless otherwise noted.

mjwitte committed:
Add version comment to ResidentialACsAndHPsperfCurves data

mjwitte commented:
@Myoldmopar I went ahead and added that comment line.  When CI comes back green will merge.
